mount/mount.go

```diff
@@ -22,3 +28,48 @@ func All(mounts []Mount, target string) error {
 	}
 	return nil
 }
+
+// MountsOption are option for mount. It indicate that we will mount Mounts to TmpParentDir/TmpDirPrefix
+// for TmpParentDir and TmpDirPrefix, please see ioutil.TempDir.
+type MountsOption struct {
+	Mounts       []Mount
+	TmpParentDir string
+	TmpDirPrefix string
+}
+
+// VisitMountedDir mounts all the provided mounts to the tmp dir, and pass the tmp dir to f.
+func VisitMountedDir(mountOption []MountsOption, f func(roots []string) error) (err error) {
+	var roots []string
+
+	if len(mountOption) == 0 {
+		return fmt.Errorf("invalid argument")
+	}
+	for _, o := range mountOption {
+		root, err := ioutil.TempDir(o.TmpParentDir, o.TmpDirPrefix)
```

Comment:
Maybe move more of this logic into your new struct so that we can call methods on it for create(), remove() { unmount;remove}, etc

Reply:
Sorry, I don't know what you mean. 
Can you tell what the create() and remove() means?  or give me a simple example.